src/metpy/plots/declarative.py

```diff
@@ -647,6 +651,28 @@ class MapPanel(Panel):
     This trait sets a user-defined title that will plot at the top center of the figure.
     """
 
+    @validate('area')
+    def _valid_area(self, proposal):
+        """Check that the proposed area string or tuple is a valid value."""
+        area = proposal['value']
+
+        # If string, check that the string exists in list of areas ('global' is special case)
+        if isinstance(area, str):
+            region = self._parse_area_string(area)[0]
+            if region.lower() not in _areas.keys() and region != 'global':
+                raise TraitError('"' + area + '" is not a valid string area.')
+        # If tuple, check that the latitudes and longitudes make sense
+        elif isinstance(area, tuple):
+            west_lon, east_lon, south_lat, north_lat = area
+            valid_west = -180 <= west_lon <= 180
+            valid_east = -180 <= east_lon <= 180
+            valid_south = -90 <= south_lat <= 90
+            valid_north = -90 <= north_lat <= 90
+            if not valid_west or not valid_east or not valid_south or not valid_north:
+                raise TraitError(str(area) + ' is not a valid latitude/longitude extent.')
+
+        return area
```

Comment:
I wonder if it would be better to just return the area as an extent since we've done most of the work here. We'd have to let through the `global` string as a way to later call `set_global`, but it would leave `draw` simpler.

Reply:
Gotcha, now I see where we're going with traitlets validation here